demisto_sdk/commands/validate/initializer.py

```diff
@@ -287,6 +360,114 @@ def get_unfiltered_changed_files_from_git(self) -> Tuple[Set, Set, Set]:
+        4. Log any processing errors without failing
+        5. Continue with validation even if no status files are found
+        """
+
+        if self.handling_private_repositories:
+            artifacts_folder = os.getenv("ARTIFACTS_FOLDER", "")
+            logs_dir = (
+                Path(artifacts_folder) / "logs" if artifacts_folder else Path("logs")
+            )
+
+            status_files = [
+                logs_dir / "content_private_files_relative_paths.txt",
+                logs_dir / "content_test_conf_files_relative_paths.txt",
+                logs_dir / "content_configuration_files_relative_paths.txt",
+            ]
+
+            for status_file in status_files:
+                try:
+                    if status_file.exists():
+                        with open(status_file, "r") as f:
+                            file_statuses = DEFAULT_JSON_HANDLER.load(f)
+                            modified_count = 0
+                            renamed_count = 0
+
+                            for file_path_str, status_info in file_statuses.items():
+                                if not file_path_str:
+                                    continue
+
+                                file_path = Path(file_path_str)
+
+                                # Handle renamed files
+                                if (
+                                    status_info.get("status") == "renamed"
+                                    and file_path in added_files
+                                ):
+                                    added_files.discard(file_path)
+                                    old_path = status_info.get("old_path")
+                                    renamed_files.add((Path(old_path), file_path))  # type: ignore[arg-type]
+                                    renamed_count += 1
+                                    continue  # Use continue instead of break to process remaining files
+
+                                # Handle modified status
+                                if (
+                                    status_info.get("status") == "modified"
+                                    and file_path in added_files
+                                ):
+                                    added_files.discard(file_path)
+                                    modified_files.add(file_path)
+                                    modified_count += 1
+
+                                # Files marked as "added" remain in added_files
+
+                except DEFAULT_JSON_HANDLER.JSONDecodeError as e:
+                    logger.warning(f"Invalid JSON format in {status_file}: {str(e)}")
+                except Exception as e:
+                    logger.warning(f"Error processing {status_file}: {str(e)}")
+                    logger.debug("Full traceback:", exc_info=True)
+                    continue
+
+        # Log files in a more readable format
```

Comment:
Extract the handling part (if it is easy) and the logging part of the data here to different functions, to make this function smaller

Reply:
sure